### ISSUE TITLE
Hide `updateState()` from end user API

### ISSUE THREAD

**Issue by alexander-yevsyukov** [SpineEventEngine/core-java#782]
`AbstractVersionableEntity` exposes `updateState()` as `public`. Even though, it has the method marked `@Internal`, there is a great risk of misusing the API via end users.

The reason for exposure is documented in `AbstractVersionableEntityTest.haveUpdateStatePackagePrivate()` as

```java
@Disabled // The `updateState` method was made public to be accessible from `testutil-server`
```

Current calls of `updateState()` from the `testutil` base are limited to a single usage in `EntityBuilder`:

```java
    protected void setState(E result, S state, Version version) {
        result.updateState(state, version);
    }
```
At the same time this particular method implementation seems to be never used, but overridden by each of its descendants, such as `AggregateBuilder`. In _their_ implementation of `setState()` no calls to `updateState()` are made.

To do:

1. Make `EntityBuilder.setState()` no-operational with the comment. Or try to make it `abstract` along with `EntityBuilder` itself — this will take some effort, but seems doable.
2. Rollback `AbstractVersionableEntity.updateState()` back to package-private.
3. Re-enable `AbstractVersionableEntityTest.haveUpdateStatePackagePrivate()` .

**Comment by alexander-yevsyukov** [comment]
@armiol, FYI.

**Comment by alexander-yevsyukov** [comment]
@vlad-lubenskyi, can this issue be closed, assuming #784 is merged?
